### PR TITLE
Add cross-concept linking to intros and abouts

### LINKED ISSUE #580: [v3] Add cross-concept linking to concept `about.md` files
Once the syntax for linking to contexts on the v3 website is finalized, let's add some cross-concept links to concept `about.md` files. E.g. here https://github.com/exercism/elixir/blob/main/concepts/case/about.md#L3 `pattern matching` and `guards` could be linked.

https://github.com/exercism/docs/blob/main/building/markdown/internal-linking.md

### PR DESCRIPTION
**Pull request by angelikatyborska**
Resolves #580

I tried to avoid linking between concepts that are taught by the same exercise, e.g. multiple clause functions and guards, processes and pids.

### CODE REVIEW

**Review by jiegillet — COMMENTED** [review]
Amazing work, was there a method to it? It looks so tedious :)

I think I spotted two missing links.

I ran a [markdown link checker](https://github.com/tcort/markdown-link-check) on the whole repo and it flagged some things. I know it's not exactly the goal of this PR, so it could be another PR I guess, but here goes:

```
FILE: concepts/try-rescue/about.md
  ERROR: 1 dead links found!
  [✖] /tracks/elixir/concepts/exceptions → Status: 400

FILE: concepts/try-rescue/introduction.md
  ERROR: 1 dead links found!
  [✖] /tracks/elixir/concepts/exceptions → Status: 400

FILE: reference/exercise-concepts/robot-simulator.md
  ERROR: 1 dead links found!
  [✖] https://github.com/exercism/elixir/blob/master/exercises/robot-simulator/example.exs → Status: 404

FILE: reference/implementing-a-concept-exercise.md
  ERROR: 2 dead links found!
  [✖] https://github.com/exercism/docs/blob/main/anatomy/tracks/concepts.md → Status: 404
  [✖] https://github.com/exercism/docs/blob/main/anatomy/tracks/concept-exercises.md → Status: 404
```

**Comment by angelikatyborska** [comment]
> Amazing work, was there a method to it? It looks so tedious :)

It was tedious... I read all of the `about.md`s and tried to judge where it mentions another concept for the first time in the document and that's not already linked.

> I ran a [markdown link checker](https://github.com/tcort/markdown-link-check) on the whole repo and it flagged some things

That's awesome! I wish we had a dead-link check on the CI for documents that don't come from problem specs.

I'll run it too and fix what I can.

**Comment by jiegillet** [comment]
> That's awesome! I wish we had a dead-link check on the CI for documents that don't come from problem specs.

This one is not perfect, I had to curat the list a little bit, it also flagged some valid links, [all resembling this one](https://github.com/pulls?q=is%3Aopen+repo%3Aexercism%2Felixir-analyzer).

File: exercises/concept/guessing-game/.meta/design.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ A student upon completion will:
 
 ## Narrative
 
-This exercise is a fork from [F#'s Number Guessing Game](https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/pattern-matching/.docs/instructions.md)
+This exercise is a fork from [F#'s Number Guessing Game](https://github.com/exercism/fsharp/tree/main/exercises/concept/guessing-game)
```
Comment (jiegillet) [inline exercises/concept/guessing-game/.meta/design.md:34]: Nice. I noticed later that my dead link check yesterday didn't include files in dotted folders.

**Review by jiegillet — APPROVED** [review]
Nice, this is a super valuable PR :)

**Pull request merged**